File: YuJin/211207/20292.py
import sys
input = sys.stdin.readline

w, r = set(), set()
result = []

while True:
    v = str(input())
    s = v.split()
    if s[0] == 'WRITE':
        if s[1] in w or s[3] in r or s[3] in w:
            r.clear()
            w.clear()
            result.append('WAIT\n')
        w.add(s[3])
        r.add(s[1])
    elif s[0] == 'READ':
        if s[1] in w:
            r.clear()
            w.clear()
            result.append('WAIT\n')
        r.add(s[1])
    else:
        break
    result.append(v)

result.append('EXIT')
print("".join(result))

File: YeoHun/211207/20292.swift
import Foundation

var sourceSet = Set<String>()
var targetSet = Set<String>()
var readSet = Set<String>()

func clear() {
    sourceSet = Set<String>()
    targetSet = Set<String>()
    readSet = Set<String>()
}

while true {
    let input = readLine()!
    
    let command = input.components(separatedBy: " ").filter({ $0 != " " && $0 != "TO" })
    if command[0] == "WRITE" {
        if readSet.contains(command[2]) {
            print("WAIT")
            clear()
        }
        else if targetSet.contains(command[1]) || sourceSet.contains(command[0]) {
            print("WAIT")
            clear()
        }
        else if targetSet.contains(command[2]) {
            print("WAIT")
            clear()
        }
        else if targetSet.contains(command[1]) && sourceSet.contains(command[2]) {
            print("WAIT")
            clear()
        }
        sourceSet.insert(command[1])
        targetSet.insert(command[2])
    }
    if command[0] == "READ" {
        if targetSet.contains(command[1]) {
            print("WAIT")
            clear()
        }
        readSet.insert(command[1])
    }
    print(input)
    
    if command[0] == "EXIT" {
        break
    }
}
